Promises callbacks js

shidalgo168 opened the pull request:
Promise excercise of the General Training, using Run Forest, Run! assignment as a base
Run Forest, Run! reference: https://www.figma.com/file/Cujgnp1yC8a3ZAOWviCaNq0g/Agency-Directory-Design?node-id=0%3A1

shidalgo168 committed:
basic files for the assignment

shidalgo168 committed:
movile styles for the webpage

shidalgo168 committed:
upload the logo font

shidalgo168 committed:
media-queries

shidalgo168 committed:
gulpfile fix and promise/httpReq search

shidalgo168 committed:
add tasks to watch in gulpfile and deletion of c.log()

shidalgo168 committed:
deleting the use of jquery

shidalgo168 committed:
pixel perfect fixes and "Enter" search function

asalazarb commented on I do magic/promises-callbacks-js/src/css/body-top.scss:
```diff
@@ -0,0 +1,74 @@
+  color: $light-text;
+  opacity: 1; /* Firefox */
+  padding-left: 25px;
+  font-family: Arial, sans-serif;
```
Usar variable para los fonts, ej:
$arial-bold: 'Arial Bold', Arial;

shidalgo168 replied:
Done

asalazarb commented on I do magic/promises-callbacks-js/src/js/load-data.js:
```diff
@@ -0,0 +1,66 @@
+function validateEnter(e) {
+  if (e.keyCode===13) getAgents('https://api.myjson.com/bins/uptto');
```
Guardar el 13 en un const: const RETURN_KEY = 13. Para que sea más entendible.

hansjag04 replied:
Igual para el URL del API

shidalgo168 replied:
Done

asalazarb commented on I do magic/promises-callbacks-js/src/js/load-data.js:
```diff
@@ -0,0 +1,66 @@
+        if(data.companies[i].name.toUpperCase().includes(userInput)){
+          newFilter.companies.push(data.companies[i]);
+        }
+      }
```
Ese filtro se puede hacer más sencillo con: https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/Array/filter

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/gulpfile.js:
```diff
@@ -0,0 +1,82 @@
+});
+
+// HTML generation task
+// var fs = require("fs");
```
Don't leave commented lines. It may confuse the development of future changes (like "is this line needed for something?", "Should I uncomment this line to build the project and then comment it?", etc). Confusing for others, and also for you.

Try not to leave them, but If you really need to, think about leaving a small comment of why that line should be there.

shidalgo168 replied:
Julio gave us that file xd

hansjag04 replied:
I would have commented the same to him 😄

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/gulpfile.js:
```diff
@@ -0,0 +1,82 @@
+// var fs = require("fs");
+// var inject = require('gulp-inject-string');
+gulp.task('html', function () {
+	// var cssContent = fs.readFileSync("./dist/main.css", "utf8");
```
Same as before (commented lines).

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/gulpfile.js:
```diff
@@ -0,0 +1,82 @@
+gulp.task('html', function () {
+	// var cssContent = fs.readFileSync("./dist/main.css", "utf8");
+	gulp.src("./src/html/*.html")
+		// .pipe(inject.after('style amp-custom>', cssContent))
```
Same as before (commented lines). If it's something that has to be done, is preferable to include a TODO comment. Be sure to include a really brief description and a task number (if you're using JIRA or Trello) with the TODO comment to avoid filling the code with lots of TODOs. -Only when necessary-.

https://medium.com/imdoneio/3-reasons-why-todo-comments-are-a-good-thing-c2cf3d7b7c2b

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/src/css/body.scss:
```diff
@@ -0,0 +1,107 @@
+  &__separator {
+    position: absolute;
+    width: 313px;
+    height: 0px;
```
No need to add `px` when value is 0.

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/src/css/footer.scss:
```diff
@@ -0,0 +1,33 @@
+.footer {
+  position: relative;
+  height: 88px;
+
```
Remove.

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/src/css/grid.scss:
```diff
@@ -0,0 +1,143 @@
+@import "_variables.scss";
+
+/*
```
Remove this block

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/src/css/grid.scss:
```diff
@@ -0,0 +1,143 @@
+  }
+  @media(min-width: 1024px) {
+    &__name {
+      left: calc(50% - 160px);
```
Is it really necessary to do all these calculations always? Whenever the window is resized, all this calculations are made. Maybe you can use a different measures based on another component instead of the whole window.
Don't need to change it, but to take in mind.

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/src/html/index.html:
```diff
@@ -0,0 +1,74 @@
+    <title>Super Agentes - Home</title>
+    <!--Bootstrap grid-->
+    <link rel="stylesheet" type="text/css" href="main.css">
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/mustache.js/2.3.2/mustache.min.js"></script>
```
Load all scripts in the bottom of the body section.

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/src/html/index.html:
```diff
@@ -0,0 +1,74 @@
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/mustache.js/2.3.2/mustache.min.js"></script>
+
+    <script type="text/x-handlebars-template" id="template">
+      {{#companies}}
```
Can you try to load this template in another file instead of using the same file? Is always better not to mix languages, or using inline scripts.

hansjag04 commented on I do magic/promises-callbacks-js/src/html/index.html:
```diff
@@ -0,0 +1,74 @@
+    <script type="text/javascript" src="js/load-data.js"></script>
+  </head>
+
+  <body language=JavaScript onLoad="getAgents('https://api.myjson.com/bins/uptto')">
```
Don't load scripts inline. Use another file and load it in this HTML. Remove the `language=JavaScript` tag.

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/src/html/index.html:
```diff
@@ -0,0 +1,74 @@
+      <div class="title">Los Mejores agentes de seguridad</div>
+      <div class="searchbar">
+          <input type="text" class="searchbar__textfield" placeholder="Buscar por Ubicación" name="Location" onkeypress="validateEnter(event)">
+          <input type="image" src="images/search.png" class="searchbar__search-button" onclick="getAgents('https://api.myjson.com/bins/uptto')">
```
Don't mix JS in HTML using onclick="method()". Use event handlers instead.

// Assign a function to deal with the onclick event:
document.getElementById('id').onclick = doSomething;

// or use an anonymous function:
document.getElementById('id').onclick = function() {
    // do magic
};

https://stackoverflow.com/questions/5871640/why-is-using-onclick-in-html-a-bad-practice

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/src/html/index.html:
```diff
@@ -0,0 +1,74 @@
+      <div class="image"></div>
+      <div class="title">Los Mejores agentes de seguridad</div>
+      <div class="searchbar">
+          <input type="text" class="searchbar__textfield" placeholder="Buscar por Ubicación" name="Location" onkeypress="validateEnter(event)">
```
Same as the next comment; Don't mix JS in HTML; is not a good practice. Use event handlers instead of `onkeypress=(...)`.

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/src/js/load-data.js:
```diff
@@ -0,0 +1,66 @@
+
+function get(url) {
+  // Return a new promise.
+  return new Promise(function(resolve, reject) {
```
Use arrow functions instead
return new Promise((resolve, reject) => { ... })

https://medium.freecodecamp.org/when-and-why-you-should-use-es6-arrow-functions-and-when-you-shouldnt-3d851d7f0b26

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/src/js/load-data.js:
```diff
@@ -0,0 +1,66 @@
+}
+
+function get(url) {
+  // Return a new promise.
```
Unnecessary comment.
https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29#comments-rules

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/src/js/load-data.js:
```diff
@@ -0,0 +1,66 @@
+    }
+    //injecting the agent-objects into the html
+    document.getElementById('agent-section').innerHTML = agents;
+  }, function(error) {
```
an arrow function looks better here.

shidalgo168 replied:
Done

shidalgo168 committed:
Reviews fixed

hansjag04 commented on I do magic/promises-callbacks-js/src/html/index.html:
```diff
@@ -58,8 +36,8 @@
       <div class="searchbar">
-          <input type="text" class="searchbar__textfield" placeholder="Buscar por Ubicación" name="Location" onkeypress="validateEnter(event)">
-          <input type="image" src="images/search.png" class="searchbar__search-button" onclick="getAgents('https://api.myjson.com/bins/uptto')">
+          <input id="searchText" type="text" class="searchbar__textfield" placeholder="    Buscar por Ubicación">
```
Why so many spaces at the beginning of the placeholder?

shidalgo168 replied:
Done

hansjag04 commented on I do magic/promises-callbacks-js/src/js/load-data.js:
```diff
@@ -1,10 +1,22 @@
+};
+
+document.getElementById('searchText').onkeypress = (event) => {
+  if (event.keyCode===ENTER_KEY) getAgents(URL_TO_RETIREVE_DATA);
```
Use spaces between operators. And prefer to use 
```
if (condition) {
    expression
}
```
than `if (condition) expression`. Or at least `if (condition) { expression }`.

shidalgo168 replied:
Ack

hansjag04 commented on I do magic/promises-callbacks-js/src/js/mustache.js:
```diff
@@ -0,0 +1,16 @@
+// Mustache template for the agents
+var TEMPLATE = '{{#companies}}\
```
Shouldn't be this a const? Also, all caps labels should be const, not vars.

hansjag04 replied:
And this should be a mustache template file, not inlined in the JS. Remember not to mix languages, is not a good practice. Ask Kristal about how she used an external mustache file.

shidalgo168 replied:
Done

hansjag04 reviewed (changes requested):
Please resolve comments. Include a better description in the PR and try to add a comment in each 'conversation', like 'Done' or 'Ack' (acknowledged).

shidalgo168 committed:
fixed the latest comments

Mustache template is now loaded as a mustache file, not as a js file (changes in the gulpfile for this)
An 'if' in load-data without brackets
And the placeholder of the textbar

jcbargo commented:
@asalazarb @shidalgo168 @hansjag04 what's the status of this PR?

hansjag04 reviewed (approved):
LGTM
